apps/api/src/domain/models/digital-text/digital-text.entity.ts

```diff
@@ -0,0 +1,107 @@
+import { InternalError, isInternalError } from '../../../lib/errors/InternalError';
+import { Maybe } from '../../../lib/types/maybe';
+import { NotFound } from '../../../lib/types/not-found';
+import { DTO } from '../../../types/DTO';
+import { ResultOrError } from '../../../types/ResultOrError';
+import formatAggregateCompositeIdentifier from '../../../view-models/presentation/formatAggregateCompositeIdentifier';
+import { buildMultilingualTextWithSingleItem } from '../../common/build-multilingual-text-with-single-item';
+import { MultilingualText } from '../../common/entities/multilingual-text';
+import { AggregateRoot } from '../../decorators';
+import { AggregateCompositeIdentifier } from '../../types/AggregateCompositeIdentifier';
+import { AggregateId } from '../../types/AggregateId';
+import { AggregateType } from '../../types/AggregateType';
+import { ResourceType } from '../../types/ResourceType';
+import { Resource } from '../resource.entity';
+import { BaseEvent } from '../shared/events/base-event.entity';
+import { CreateDigitalText } from './commands';
+
+@AggregateRoot(AggregateType.digitalText)
+@RegisterIndexScopedCommands(['CREATE_DIGITAL_TEXT'])
+export class DigitalText extends Resource {
+    readonly type = ResourceType.digitalText;
+
+    @NestedDataType(MultilingualText, {
+        label: 'title',
+        description: 'the title of the digital text',
+    })
+    readonly title: MultilingualText;
+
+    constructor(dto: DTO<DigitalText>) {
+        super({ ...dto, type: ResourceType.digitalText });
+
+        if (!dto) return;
+
+        const { title } = dto;
+
+        this.title = new MultilingualText(title);
+    }
+
+    getName(): MultilingualText {
+        return this.title.clone();
+    }
+
+    protected getResourceSpecificAvailableCommands(): string[] {
+        return [];
+    }
+
+    protected validateComplexInvariants(): InternalError[] {
+        return [];
+    }
+
+    static fromEventHistory(
+        eventStream: BaseEvent[],
+        idOfDigitalTextToCreate: AggregateId
+    ): Maybe<ResultOrError<DigitalText>> {
+        const eventsForThisDigitalText = eventStream.filter(({ payload }) =>
+            isDeepStrictEqual((payload as ICommandBase)[AGGREGATE_COMPOSITE_IDENTIFIER], {
+                type: AggregateType.digitalText,
+                id: idOfDigitalTextToCreate,
+            })
+        );
+
+        if (eventsForThisDigitalText.length === 0) return NotFound;
+
+        const [creationEvent, ...updateEvents] = eventsForThisDigitalText;
+
+        if (creationEvent.type !== `DIGITAL_TEXT_CREATED`) {
+            throw new InternalError(
+                `The first event for ${formatAggregateCompositeIdentifier({
+                    type: AggregateType.digitalText,
+                    id: idOfDigitalTextToCreate,
+                })} should have been of type DIGITAL_TEXT, but found: ${creationEvent?.type}`
```

Comment:
Do you mean to interpolate `${DIGITAL_TEXT_CREATED}` from the constant that holds the event type (you need to export this from a constants file first, though)?

Reply:
This has been done.